File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/EpisodeOfCare.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type EpisodeOfCareDiagnosis = BackboneElement & {
    rank?: number;
    role?: CodeableConcept;
    condition: Reference;
};

export type EpisodeOfCareStatusHistory = BackboneElement & {
    period: Period;
    status: string;
};

export type EpisodeOfCare = DomainResource & {
    patient: Reference;
    diagnosis?: EpisodeOfCareDiagnosis[];
    managingOrganization?: Reference;
    type?: CodeableConcept[];
    account?: Reference[];
    referralRequest?: Reference[];
    team?: Reference[];
    status: string;
    identifier?: Identifier[];
    period?: Period;
    careManager?: Reference;
    statusHistory?: EpisodeOfCareStatusHistory[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/EpisodeOfCare.py
from typing import Optional, List
from pydantic import *
from ..base import *

class EpisodeOfCare_Diagnosis(BackboneElement):
    rank: Optional[int] = None
    role: Optional[CodeableConcept] = None
    condition: Reference

class EpisodeOfCare_StatusHistory(BackboneElement):
    period: Period
    status: str

class EpisodeOfCare(DomainResource):
    patient: Reference
    diagnosis: Optional[List[EpisodeOfCare_Diagnosis]] = None
    managing_organization: Optional[Reference] = None
    type: Optional[List[CodeableConcept]] = None
    account: Optional[List[Reference]] = None
    referral_request: Optional[List[Reference]] = None
    team: Optional[List[Reference]] = None
    status: str
    identifier: Optional[List[Identifier]] = None
    period: Optional[Period] = None
    care_manager: Optional[Reference] = None
    status_history: Optional[List[EpisodeOfCare_StatusHistory]] = None